### ISSUE TITLE
Movement is canceling via by full cells.

### ISSUE THREAD

**Issue by AoAGeneral** [OpenRA/OpenRA#14073]
Currently in the new playtest (20170923) units are not moving to the closest cells to their destinations. Instead what is happening is when mobs of infantry occupy the cells that those units are trying to reach as their destination all movement orders cancels. This also includes those infantry units on the move to that spot and creates a line of units back behind the group.

To re-create this issue have a decent sized army of any unit type. Have vehicles on one key and infantry on another key. Move them to the same cell locations and this will trigger the issue.

This also affects RA. In TD its noticable as well but for some reason it seems like attack move is canceling out as well. I haven't been able to reproduce this just yet.

EDIT: This also seems to have an issue with ore trucks and harvesters temporarily canceling out when that dump area is blocked by another harvester/ore truck.

**Comment by netnazgul** [comment]
A bit of elaboration on that (if it's the same issue).
In our game, I had some observations on bugged behaviour:

- units tend to alternate attacking and moving when under A-move command (my V2s fired their shot, then moved one cell, then tried to fire with no ammo, moved again etc.)
- red lines act weird - some units showed their target on an a-move point but specific moving units at the back of enemy blob which they tried to reach ignoring closer targets

**Comment by pchote** [comment]
Can you please elaborate on the regression here, ideally with specific steps to reproduce?  This doesn't sound different to how attack move and blocked move orders have always behaved.

**Comment by AoAGeneral** [comment]
I did some games vs bots with debug mode.

The movement behavior reported here is as @pchote explained. So this has been around for a very long time. To reproduce this get a large amount of infantry and move them around on the map with right clicking only. They will trail behind. With attack move you don't see this happening. They will all move to that cell and try to fill all cells closest to that destination.

With the movement they will fill the cells at the destination and then the rest of the infantry will stop. So this is correct behavior and this is my error for not realizing this sooner. Sorry @_@

@netnazgul this is correct. I also noticed this doing the same close looks. Infantry on defense stance will have attack red lines pop up trying to attack the units. So see this happen have your units on a number key. Let them get hit by long range units (IE: Rocket infantry). With the army selected hold down the ALT key and you will see them trying to attack even out of range.
